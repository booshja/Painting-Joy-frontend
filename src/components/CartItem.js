// dependencies
import React, { useContext } from "react";
import styled from "styled-components";
// context
import CartContext from "../context/CartContext";
// breakpoints
import { breakpoints } from "../breakpoints";

const StyledCartItem = styled.div`
    display: flex;
    align-items: center;
    padding: 1rem 0;
`;

const StyledImg = styled.img`
    width: 40%;
    margin-right: 1rem;
`;

const StyledInfoDiv = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    position: relative;
`;

const StyledText = styled.p`
    font-family: "News Cycle", sans-serif;
    font-size: 1rem;
    letter-spacing: 1px;
    margin-bottom: 0.5rem;

    ${breakpoints("font-size", "rem", [{ 414: 1.2 }])}
`;

const StyledTitle = styled.p`
    font-family: "Barlow Condensed", sans-serif;
    font-size: 1.2rem;
    font-weight: 700;
    letter-spacing: 1px;
    margin-bottom: 1rem;

    ${breakpoints("font-size", "rem", [{ 414: 1.4 }])}
`;

const StyledRemove = styled.button`
    font-family: "News Cycle", sans-serif;
    font-size: 1rem;
    font-weight: 700;
    letter-spacing: 1px;
    color: #207070;
    border: none;
    background-color: transparent;
    padding: 0;

    ${breakpoints("font-size", "rem", [{ 414: 1.2 }])}
`;

const CartItem = ({ item }) => {
    // set up context
    const { removeFromCart } = useContext(CartContext);

    const handleClick = () => {
        // when clicked, remove item from cart
        removeFromCart(item.id);
    };

    return (
        <StyledCartItem>
            <StyledImg
                src={
                    process.env.REACT_APP_BACKEND_URL +
                    `items/item/${item.id}/image`
                }
                alt={item.name}
            />
            <StyledInfoDiv>
                <StyledTitle>{item.name}</StyledTitle>
                <StyledText>Price: ${item.price}</StyledText>
                <StyledText>Shipping: ${item.shipping}</StyledText>
                <StyledRemove onClick={handleClick}>Remove</StyledRemove>
            </StyledInfoDiv>
        </StyledCartItem>
    );
};

export default CartItem;
