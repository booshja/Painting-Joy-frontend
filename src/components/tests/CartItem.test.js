import { render } from "@testing-library/react";
import CartItem from "../CartItem";
import CartContext from "../../context/CartContext";
import "jest-styled-components";

/** Smoke Test */
it("renders without crashing", () => {
    render(
        <CartContext.Provider value={{ removeFromCart: null }}>
            <CartItem
                item={{
                    id: 1,
                    name: "Name",
                    price: "199.99",
                    shipping: "123.99",
                }}
            />
        </CartContext.Provider>
    );
});

/** Snapshot Test */
it("matches snapshot", () => {
    const { asFragment } = render(
        <CartContext.Provider value={{ removeFromCart: null }}>
            <CartItem
                item={{
                    id: 1,
                    name: "Name",
                    price: "199.99",
                    shipping: "123.99",
                }}
            />
        </CartContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly", () => {
    const { getByText, getByAltText, getByRole } = render(
        <CartContext.Provider value={{ removeFromCart: null }}>
            <CartItem
                item={{
                    id: 1,
                    name: "Name",
                    price: "199.99",
                    shipping: "123.99",
                }}
            />
        </CartContext.Provider>
    );

    expect(getByAltText("Name")).toBeInTheDocument();
    expect(getByText("Name")).toBeInTheDocument();
    expect(getByText("Price: $199.99")).toBeInTheDocument();
    expect(getByText("Shipping: $123.99")).toBeInTheDocument();
    expect(getByRole("button", { name: "Remove" })).toBeInTheDocument();
});
