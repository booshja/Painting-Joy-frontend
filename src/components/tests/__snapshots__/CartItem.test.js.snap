// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem 0;
}

.c1 {
  width: 40%;
  margin-right: 1rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  position: relative;
}

.c3 {
  font-family: "Barlow Condensed",sans-serif;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  margin-bottom: 1rem;
}

.c4 {
  font-family: "News Cycle",sans-serif;
  font-size: 1rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  margin-bottom: 0.5rem;
}

.c5 {
  font-family: "News Cycle",sans-serif;
  font-size: 1rem;
  font-weight: 700;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  color: #207070;
  border: none;
  background-color: transparent;
  padding: 0;
}

@media screen and (min-width:414px) {
  .c3 {
    font-size: 1.4rem;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    font-size: 2rem;
  }
}

@media screen and (min-width:1024px) {
  .c3 {
    font-size: 1.6rem;
  }
}

@media screen and (min-width:414px) {
  .c4 {
    font-size: 1.2rem;
  }
}

@media screen and (min-width:768px) {
  .c4 {
    font-size: 1.6rem;
  }
}

@media screen and (min-width:1024px) {
  .c4 {
    font-size: 1.4rem;
  }
}

@media screen and (min-width:414px) {
  .c5 {
    font-size: 1.2rem;
  }
}

@media screen and (min-width:768px) {
  .c5 {
    font-size: 1.6rem;
  }
}

@media screen and (min-width:1024px) {
  .c5 {
    font-size: 1.4rem;
  }
}

<div
    class="c0"
  >
    <img
      alt="Name"
      class="c1"
      src="http://localhost:5000/items/item/1/image"
    />
    <div
      class="c2"
    >
      <p
        class="c3"
      >
        Name
      </p>
      <p
        class="c4"
      >
        Price: $199.99
      </p>
      <p
        class="c4"
      >
        Shipping: $123.99
      </p>
      <button
        class="c5"
      >
        Remove
      </button>
    </div>
  </div>
</DocumentFragment>
`;
